planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp

```diff
@@ -533,11 +530,38 @@ PathWithLaneId StartPlannerModule::generateStopPath() const
 
   // generate drivable area
   const auto target_drivable_lanes = utils::getNonOverlappingExpandedLanes(
-    path, status_.lanes, planner_data_->drivable_area_expansion_parameters);
+    path, generateDrivableLanes(path), planner_data_->drivable_area_expansion_parameters);
 
   return path;
 }
 
+lanelet::ConstLanelets StartPlannerModule::getPathLanes(const PathWithLaneId & path) const
+{
+  lanelet::ConstLanelets path_lanes;
+
+  const auto & lanelet_layer = planner_data_->route_handler->getLaneletMapPtr()->laneletLayer;
+
+  std::vector<lanelet::Id> lane_ids;
+  for (const auto & p : path.points) {
+    for (const auto & id : p.lane_ids) {
+      if (std::find(lane_ids.begin(), lane_ids.end(), id) == lane_ids.end()) {
+        lane_ids.push_back(id);
+      }
+    }
+  }
+  for (const auto & id : lane_ids) {
+    path_lanes.push_back(lanelet_layer.get(id));
+  }
+
+  return path_lanes;
```

Comment:
```suggestion
  std::vector<lanelet::Id> lane_ids;
  for (const auto & p : path.points) {
    for (const auto & id : p.lane_ids) {
      if (std::find(lane_ids.begin(), lane_ids.end(), id) == lane_ids.end()) {
        lane_ids.push_back(id);
      }
    }
  }

  const auto & lanelet_layer = planner_data_->route_handler->getLaneletMapPtr()->laneletLayer;
  
  lanelet::ConstLanelets path_lanes;
  path_lanes.reserve(lane_ids.size());
  for (const auto & id : lane_ids) {
    path_lanes.push_back(lanelet_layer.get(id));
  }

  return path_lanes;
```

Before `lane_ids` are found, `lanelet_layer` and `path_lanes` is not used. So it might be more suitable to declare both variables after `lane_ids` loop. Also I suggest placing `reserve`.

Reply:
thanks it is better!